### PR TITLE
CA-239891: Fix management_device on Pool.eject

### PR DESCRIPTION
**Pull request by sharady**
Signed-off-by: Sharad Yadav <sharad.yadav@citrix.com>

### CODE REVIEW

File: ocaml/xapi/xapi_pool.ml [hunk 1]
```diff
@@ -909,11 +909,21 @@ let eject ~__context ~host =
...
     let pif = Db.PIF.get_record ~__context ~self:management_pif in
     let management_device =
-      (* assumes that the management interface is either physical or a bond or a vlan *)
+      (* Assumes that the management interface is either physical or a bond or a vlan on physical or a vlan on bond *)
       if pif.API.pIF_bond_master_of <> [] then
         let bond = List.hd pif.API.pIF_bond_master_of in
         let primary_slave = Db.Bond.get_primary_slave ~__context ~self:bond in
         Db.PIF.get_device ~__context ~self:primary_slave
+      (* If management on VLAN on a bond or physical NIC *)
+      else if pif.API.pIF_VLAN_master_of <> Ref.null then
```
Comment (lindig) [inline ocaml/xapi/xapi_pool.ml:918]: I was expecting that the elements of `pif.API.pIF_bond_master_of` would be used here because this list is not empty.

**Review by lindig — COMMENTED** [review]
This looks clean but I would suggest @robhoes takes a look. Could the commit subject be edited such that it makes sense and does not bleed into the body of the commit message?

File: ocaml/xapi/xapi_pool.ml (hunk 1 above)
Comment (gaborigloi) [reply ocaml/xapi/xapi_pool.ml:918]: I think `pif.API.pIF_bond_master_of` is used in the above branch, where it's non-empty, in the subsequent branches it is empty, and not used. In this branch, `pif.API.pIF_VLAN_master_of` is used, which is non-null in this branch, so I think it's fine.

**Review by robhoes — APPROVED** [review]

**Comment by sharady** [comment]
Thanks for the review :)

**Review by lindig — APPROVED** [review]

**Pull request merged**